src/OpenTK.Windowing.Desktop/NativeWindow.cs

```diff
@@ -50,6 +50,27 @@ public class NativeWindow : INativeWindow
         /// </summary>
         private double _lastMousePositionY;
 
+        private WindowIcon _icon;
+
+        /// <inheritdoc />
+        public WindowIcon Icon
+        {
+            get => _icon;
+            set
+            {
+                unsafe
+                {
+                    fixed (GraphicsLibraryFramework.Image* ptr =
+                        Unsafe.As<GraphicsLibraryFramework.Image[]>(value.Images))
```

Comment:
That won't work as they are differentyl laid out in memory. 
GraphicsLibraryFramework.Image[] has aligned structs one after another in memory, while WindowIcon.Images has Image-class references one after another in memory.

Reply:
Well, that sucks.

Should I turn Image into a struct instead, since that's a value type?

Reply:
Even then the memory layout needs to be the same which we can not necessarily guarantee, as GLFW changes and OpenTK does not or vice versa. And even if the memory layout is the same we will probably need to do extra work on other platforms like Android/iOS, so i'm not too sure what would be best. I don't like all that copying either, so perhaps wait for some other opinions.